### PR TITLE
Create ObjectEntryIndex

### PR DESCRIPTION
**Pull request by Gymnasiast**
Continues where #10966 left off. Partly backported from #10664.

### CODE REVIEW

File: src/openrct2/object/Object.h [hunk 1]
```diff
@@ -265,7 +266,7 @@ extern int32_t object_entry_group_encoding[];
 bool object_entry_is_empty(const rct_object_entry* entry);
 bool object_entry_compare(const rct_object_entry* a, const rct_object_entry* b);
 int32_t object_calculate_checksum(const rct_object_entry* entry, const void* data, size_t dataLength);
-bool find_object_in_entry_group(const rct_object_entry* entry, uint8_t* entry_type, uint8_t* entry_index);
+bool find_object_in_entry_group(const rct_object_entry* entry, uint8_t* entry_type, uint16_t* entryIndex);
```
Comment (Gymnasiast) [inline src/openrct2/object/Object.h]: ObjectEntryIndex

**Comment by IntelOrca** [comment]
I'd rather this was done after https://github.com/OpenRCT2/OpenRCT2/pull/10967 is merged.

**Comment by Gymnasiast** [comment]
@IntelOrca I'll rebase this once #10967 is merged.

File: src/openrct2/object/ObjectList.cpp [hunk 2]
```diff
@@ -135,13 +135,13 @@ void get_type_entry_index(size_t index, uint8_t* outObjectType, uint16_t* outEnt
     if (outObjectType != nullptr)
         *outObjectType = objectType;
     if (outEntryIndex != nullptr)
-        *outEntryIndex = (uint8_t)index;
+        *outEntryIndex = (ObjectEntryIndex)index;
```
Comment (tupaschoal) [inline src/openrct2/object/ObjectList.cpp]: Prefer `static_cast`

File: src/openrct2/rct1/S4Importer.cpp [hunk 3]
```diff
@@ -117,18 +119,18 @@ class S4Importer final : public IParkImporter
...
-    uint16_t _pathAdditionTypeToEntryMap[16]{};
-    uint16_t _sceneryThemeTypeToEntryMap[24]{};
+    ObjectEntryIndex _rideTypeToRideEntryMap[RCT1_RIDE_TYPE_COUNT]{};
+    ObjectEntryIndex _vehicleTypeToRideEntryMap[RCT1_VEHICLE_TYPE_COUNT]{};
+    ObjectEntryIndex _smallSceneryTypeToEntryMap[256]{};
+    ObjectEntryIndex _largeSceneryTypeToEntryMap[256]{};
+    ObjectEntryIndex _wallTypeToEntryMap[256]{};
+    ObjectEntryIndex _pathTypeToEntryMap[24]{};
+    ObjectEntryIndex _pathAdditionTypeToEntryMap[16]{};
+    ObjectEntryIndex _sceneryThemeTypeToEntryMap[24]{};
```
Comment (tupaschoal) [inline src/openrct2/rct1/S4Importer.cpp]: Should we convert all these hardcoded numbers to constants in the future?

Comment (Gymnasiast) [reply src/openrct2/rct1/S4Importer.cpp]: Not sure. They're local and only used for RCT1 import.

**Pull request merged**